src/H5Oattr.c

```diff
@@ -234,17 +268,18 @@ H5O__attr_decode(H5F_t *f, H5O_t *open_oh, unsigned H5_ATTR_UNUSED mesg_flags, u
     if ((attr->shared->data_size / dt_size) != ds_size)
         HGOTO_ERROR(H5E_RESOURCE, H5E_OVERFLOW, NULL, "data size exceeds addressable range")
 
-    /* Go get the data */
+    /* Get the data */
     if (attr->shared->data_size) {
         /* Ensure that data size doesn't exceed buffer size, in case of
-           it's being corrupted in the file */
-        if (attr->shared->data_size > p_size)
-            HGOTO_ERROR(H5E_RESOURCE, H5E_OVERFLOW, NULL, "data size exceeds buffer size")
```

Comment:
This check was incorrect and would not catch cases where the data size was too big (and could run off the end of the buffer) but less than the size of the entire message.